include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -1110,34 +1110,35 @@ namespace VectorTools
   {
     const unsigned int fe_degree = matrix_free.get_dof_handler().get_fe().degree;
 
+    (void)n_q_points_1d;
     Assert (fe_degree+1 == n_q_points_1d,
             ExcNotImplemented());
 
     switch (fe_degree)
       {
       case 1:
-        project_parallel<dim, VectorType,1,2> (matrix_free,constraints,func,vec_result);
+        project_parallel<dim, VectorType, dim, 2> (matrix_free,constraints,func,vec_result);
```

Comment:
I saw a lot of compilation errors of the form

```cpp
/home/wellsd2/Documents/Code/CPP/dealii-dev/include/deal.II/grid/tria_boundary.h:147:3:
error: no type named ‘face_iterator’ in ‘class dealii::Triangulation<3, 8>’
   get_intermediate_points_on_face (const typename Triangulation<dim,spacedim>::face_iterator &face,
   ^
```

so my compiler tries to stick this on the wrong `project_parallel` function, I think.